src/Clock.tsx

```diff
@@ -0,0 +1,59 @@
+import React from 'react';
+
+type Props = {
+  clockName: string,
+};
+
+type State = {
+  today: Date;
+};
+
+export class Clock extends React.PureComponent<Props, State> {
+  state = {
+    today: new Date(),
+  };
+
+  interval = 0;
+
+  componentDidMount() {
+    this.interval = window.setInterval(() => {
+      this.setState({ today: new Date() });
+
+      // eslint-disable-next-line no-console
+      console.info(this.state.today.toUTCString().slice(-12, -4));
+    }, 1000);
+  }
+
+  componentDidUpdate(prevProps: Props) {
+    const newName = this.props.clockName;
+    const oldName = prevProps.clockName;
+
+    if (oldName !== newName) {
+      // eslint-disable-next-line no-console
+      console.debug(`Renamed from ${oldName} to ${newName}`);
+    }
+  }
+
+  componentWillUnmount() {
+    clearInterval(this.interval);
+  }
+
+  render() {
+    const { today } = this.state;
+    const { clockName } = this.props;
+
+    return (
+      <div className="Clock">
+        <strong className="Clock__name">
+          {clockName}
+        </strong>
+
+        {' time is '}
+
+        <span className="Clock__time">
+          {today.toUTCString().slice(-12, -4)}
```

Comment:
It is better to put such calculations in separate variables to improve readability.